universal/include/userver/utils/ip/address_base.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <array>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace utils::ip::detail {
+
+template <std::size_t N>
+constexpr bool kIsValidAddressSize = (N == 4 || N == 16);
+
+/// @ingroup userver_containers
+///
+/// @brief Base class for IPv4/IPv6 addresses
+template <std::size_t N, typename = std::enable_if_t<kIsValidAddressSize<N>>>
+class AddressBase final {
```

Comment:
If members of this type (e.g. the constructor and `GetBytes`) should be touched by the user, then this type should not be in `details` or `impl`